stdlib/log-api/src/main/java/org/ballerinalang/stdlib/log/Utils.java

```diff
@@ -27,39 +27,76 @@
  * @since 1.1.0
  */
 public class Utils extends AbstractLogFunction {
+    private static boolean logLevelEnabled = false;
+
     public static void printDebug(Object msg) {
-        logMessage(Scheduler.getStrand(), msg, BLogLevel.DEBUG, getPackagePath(),
-                (pkg, message) -> {
-            getLogger(pkg).debug(message);
-        });
+        if (LOG_MANAGER.isModuleLogLevelEnabled()) {
+            logLevelEnabled = isLogLevelEnabled(BLogLevel.DEBUG, getPackagePath());
+        } else {
+            logLevelEnabled = isLogLevelEnabled(BLogLevel.DEBUG, "");
+        }
+        if (logLevelEnabled) {
+            logMessage(Scheduler.getStrand(), msg, BLogLevel.DEBUG, getPackagePath(),
```

Comment:
As mentioned before, it makes a considerable performance improvement when the log level is not enabled. With the new implementation, if module log level is not enabled, `getPackagePath()` does not have to get called in order to check if the log level is enabled. If only the log level is enabled, `getPackagePath()` will be called to print the log message.

Yes, agree with the second point. the check in logMessage() is not redundant since it is already confirmed that the log level is enabled. Hence removed that check.

Reply:
Sorry, yeah, you are right, no need to be doing additional evaluations if the log level is not enabled